### PR TITLE
setup prettier

### LINKED ISSUE #30: dev: Set up Prettier
- At least for JS files. We could explore adding it for markdown files too.
- Config file could be similar to the SE-2 one for consistency.

### PR DESCRIPTION
**Pull request by technophile-04**
### Description : 

Used [.prettierrc.json](https://github.com/scaffold-eth/scaffold-eth-2/blob/main/packages/nextjs/.prettierrc.json) from SE-2 

Also prettier by default nicely formats `.md` and intelligently formats code blocks inside `.md` we could ignore `.md` files but I think it's nice to have it for formatting tables etc

Also, should we set up eslint and gh-actions too ? 

Fixes #30

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "clear": "docusaurus clear",
     "serve": "docusaurus serve",
     "write-translations": "docusaurus write-translations",
-    "write-heading-ids": "docusaurus write-heading-ids"
+    "write-heading-ids": "docusaurus write-heading-ids",
+    "format": "npx prettier --write ."
```
Comment (carletex) [inline package.json:15]: A couple of ideas here:

1. Should we do `prettier --write '**/*.{js,md,mdx,json}'` so we are a bit more clear on what we want to format?
2. Should we add a format-check (`prettier --check '**/*.{js,md,mdx,json}'`)? Could be used on a GH action.

**Review by carletex — COMMENTED** [review]
Thanks you @technophile-04 ! And so happy that you made different commits so prettified files doesn't pollute on review ;)

> Also, should we set up eslint and gh-actions too ?

- I think eslint is not needed since most JS files are config files
- I'd be up for GH action that checks the prettier formatting on PRs.

Regarding md/mdx files, found this:

https://docusaurus.io/docs/markdown-features/admonitions

Tried it, but it doesn't seem to be an issue me. But just wanted to mention it here.

**Comment by technophile-04** [comment]
> - I think eslint is not needed since most JS files are config files

💯💯


> - I'd be up for GH action that checks the prettier formatting on PRs.
> - Should we add a format-check (prettier --check '**/*.{js,md,mdx,json}')? Could be used on a GH action

Yup added it at [9645f57](https://github.com/scaffold-eth/se2-docs/pull/31/commits/9645f575325bc215709ccd456d9e21a6dd482d83)

> Regarding md/mdx files, found this:
>https://docusaurus.io/docs/markdown-features/admonitions
>Tried it, but it doesn't seem to be an issue me. But just wanted to mention it here.

Yeah lol even I noticed that it didn't messed up admonitions even if we don't have extra line spacing 

Doing some research I found that if you run `npx prettier . --write --prose-wrap always` it messes up the admonitions notice `--prose-wrap always` causes it. 

Checkout out [Prose Wrap option here](https://prettier.io/docs/en/options.html#prose-wrap), related discussion https://github.com/prettier/prettier/issues/9154

Since by default `prose-wrap` is not applied to markdown files, that's why I think its not causing any problem for us

**Comment by carletex** [comment]
Great stuff @technophile-04 !! 

I also added CSS to the prettier list, since I saw `custom.css` being fixed :)

@edakturk14 @Pabl0cks @ZakGriffith If your IDE doesn't do it automatically, you can configure it to use prettier now.

**Comment by Pabl0cks** [comment]
> If your IDE doesn't do it automatically, you can configure it to use prettier now.

Changed to prettier for css files and is working great, ty!

**Pull request merged**